### ISSUE TITLE
[🐛 Bug]: getText() doesn't work when switching to another browser window on v.9

### ISSUE THREAD

**Issue by vvoldi** [webdriverio/webdriverio#13436]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

9.0.7

### Node.js Version

21.6.2

### Mode

Standalone Mode

### Which capabilities are you using?

```typescript
{
  browserName: 'chrome',
}
```

### What happened?

after switching to v.9*, the `.getText()` method on a tab opened with `browser.newWindow()` => `browser.switchWindow()` stopped working in wdio 

### What is your expected behavior?

the user can get the text from the element on the newly switched tab using the methods `browser.newWindow()` => `browser.switchWindow()` => `$().getText()`

### How to reproduce the bug.

1. create a new test environment with `npm init wdio@latest ./`
2. press “Enter” to all questions during creation
3. open `<yourFolder>\specs\test.e2e.js` file
4. add after first `it()` block this it block: 
```
it.only("should switch to new window and get text", async () => {
        await browser.url("https://google.com");
        await browser.newWindow("https://example.com");
        await browser.switchWindow("example.com");
        const text = await $("h1").getText();
        console.log(text);
        await browser.closeWindow();
        await browser.switchWindow("google.com");
    });
```
5. run `npm run wdio`

### Relevant log output

```typescript
$ npm run wdio

> wdio
> wdio run ./wdio.conf.js


Execution of 1 workers started at 2024-08-21T19:45:42.157Z

2024-08-21T19:45:42.177Z INFO @wdio/cli:launcher: Run onPrepare hook
2024-08-21T19:45:42.178Z INFO @wdio/utils: Setting up browser driver for: chrome@stable
2024-08-21T19:45:42.179Z INFO @wdio/utils: Setting up browser binaries for: chrome@stable
2024-08-21T19:45:42.183Z INFO webdriver: Using Chromedriver v127.0.6533.120 from cache directory C:\Users\user\AppData\Local\Temp
2024-08-21T19:45:42.187Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2024-08-21T19:45:42.188Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./wdio.conf.js
[0-0] 2024-08-21T19:45:44.102Z INFO @wdio/local-runner: Run worker command: run
[0-0] RUNNING in chrome - file:///C:/Users/user/Desktop/test/test/specs/test.e2e.js
[0-0] 2024-08-21T19:45:44.281Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2024-08-21T19:45:44.292Z INFO webdriver: Using Chromedriver v127.0.6533.120 from cache directory C:\Users\user\AppData\Local\Temp
[0-0] 2024-08-21T19:45:44.328Z INFO chromedriver: Starting ChromeDriver 127.0.6533.120 (c51f2916e4a14157c276fd33c8bfb24155aac1ad-refs/branch-heads/6533@{#2000}) on port 49642
[0-0] 2024-08-21T19:45:44.329Z INFO chromedriver: Remote connections are allowed by an allowlist (0.0.0.0).
[0-0] 2024-08-21T19:45:44.329Z INFO chromedriver: Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
[0-0] 2024-08-21T19:45:44.332Z INFO chromedriver: ChromeDriver was started successfully.
[0-0] 2024-08-21T19:45:45.328Z INFO @wdio/utils: Started Chromedriver v127.0.6533.120 with params --port=49642 --allowed-origins=* --allowed-ips=0.0.0.0 in 1045ms on port 49642
[0-0] 2024-08-21T19:45:45.367Z INFO webdriver: [POST] http://localhost:49642/session
[0-0] 2024-08-21T19:45:45.368Z INFO webdriver: DATA {
[0-0]   capabilities: {
[0-0]     alwaysMatch: {
[0-0]       browserName: 'chrome',
[0-0]       'goog:chromeOptions': [Object],
[0-0]       webSocketUrl: true
[0-0]     },
[0-0]     firstMatch: [ {} ]
[0-0]   },
[0-0]   desiredCapabilities: {
[0-0]     browserName: 'chrome',
[0-0]     'goog:chromeOptions': {
[0-0]       binary: 'C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe'
[0-0]     },
[0-0]     webSocketUrl: true
[0-0]   }
[0-0] }
[0-0] 2024-08-21T19:45:46.103Z INFO webdriver: Register BiDi handler for session with id 6460d5231db5a019094c191847906560
[0-0] 2024-08-21T19:45:46.104Z INFO webdriver: Connect to webSocketUrl ws://127.0.0.1:49642/session/6460d5231db5a019094c191847906560
[0-0] 2024-08-21T19:45:46.116Z INFO webdriver: Connected session to Bidi protocol
[0-0] 2024-08-21T19:45:46.116Z INFO webdriver: Connected session to Bidi protocol
[0-0] 2024-08-21T19:45:46.117Z INFO webdriver: Connected to WebDriver Bidi interface at ws://127.0.0.1:49642/session/6460d5231db5a019094c191847906560
[0-0] 2024-08-21T19:45:46.118Z INFO webdriver: BIDI COMMAND script.addPreloadScript { functionDeclaration: <PreloadScript[868 bytes]>, contexts: undefined }
[0-0] 2024-08-21T19:45:46.120Z INFO webdriver: BIDI COMMAND session.subscribe {"events":["browsingContext.navigationStarted","network.responseCompleted","network.beforeRequestSent","network.fetchError"]}
[0-0] 2024-08-21T19:45:46.121Z INFO webdriver: BIDI COMMAND session.subscribe {"events":["browsingContext.userPromptOpened"]}
[0-0] 2024-08-21T19:45:46.121Z INFO webdriver: BIDI COMMAND session.subscribe {"events":["browsingContext.navigationStarted"]}
[0-0] 2024-08-21T19:45:46.121Z INFO webdriver: BIDI COMMAND session.subscribe {"events":["log.entryAdded"]}
[0-0] 2024-08-21T19:45:46.141Z INFO webdriver: COMMAND getWindowHandle()
[0-0] 2024-08-21T19:45:46.141Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window
[0-0] 2024-08-21T19:45:46.146Z INFO webdriver: RESULT 7C780A43FF9E5FEB981EFF9A6B2048E7
[0-0] 2024-08-21T19:45:46.147Z INFO webdriver: BIDI COMMAND browsingContext.navigate {"context":"7C780A43FF9E5FEB981EFF9A6B2048E7","url":"https://google.com","wait":"complete"}
[0-0] 2024-08-21T19:45:46.154Z INFO webdriver: COMMAND getWindowHandle()
[0-0] 2024-08-21T19:45:46.154Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window
[0-0] 2024-08-21T19:45:46.158Z INFO webdriver: RESULT 7C780A43FF9E5FEB981EFF9A6B2048E7
[0-0] 2024-08-21T19:45:46.159Z INFO webdriverio:ContextManager: Update current context: 7C780A43FF9E5FEB981EFF9A6B2048E7
[0-0] 2024-08-21T19:45:48.256Z INFO webdriver: COMMAND getWindowHandles()
[0-0] 2024-08-21T19:45:48.256Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window/handles
[0-0] 2024-08-21T19:45:48.262Z INFO webdriver: RESULT [ '7C780A43FF9E5FEB981EFF9A6B2048E7' ]
[0-0] 2024-08-21T19:45:48.263Z INFO webdriver: BIDI COMMAND browsingContext.create {"type":"window"}
[0-0] 2024-08-21T19:45:48.346Z INFO webdriver: BIDI COMMAND browsingContext.navigate {"context":"55598093E4652C509BEB5F7F0E406191","url":"https://example.com"}
[0-0] 2024-08-21T19:45:48.352Z INFO webdriver: COMMAND getWindowHandle()
[0-0] 2024-08-21T19:45:48.353Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window
[0-0] 2024-08-21T19:45:48.359Z INFO webdriver: RESULT 7C780A43FF9E5FEB981EFF9A6B2048E7
[0-0] 2024-08-21T19:45:49.346Z INFO webdriver: COMMAND getWindowHandles()
[0-0] 2024-08-21T19:45:49.346Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window/handles
[0-0] 2024-08-21T19:45:49.370Z INFO webdriver: RESULT [
[0-0]   '7C780A43FF9E5FEB981EFF9A6B2048E7',
[0-0]   '55598093E4652C509BEB5F7F0E406191'
[0-0] ]
[0-0] 2024-08-21T19:45:49.370Z INFO webdriver: COMMAND getWindowHandles()
[0-0] 2024-08-21T19:45:49.370Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window/handles
[0-0] 2024-08-21T19:45:49.375Z INFO webdriver: RESULT [
[0-0]   '7C780A43FF9E5FEB981EFF9A6B2048E7',
[0-0]   '55598093E4652C509BEB5F7F0E406191'
[0-0] ]
[0-0] 2024-08-21T19:45:49.377Z INFO webdriver: COMMAND switchToWindow("<Screenshot[base64]>")
[0-0] 2024-08-21T19:45:49.377Z INFO webdriver: [POST] http://localhost:49642/session/6460d5231db5a019094c191847906560/window
[0-0] 2024-08-21T19:45:49.377Z INFO webdriver: DATA { handle: '55598093E4652C509BEB5F7F0E406191' }
[0-0] 2024-08-21T19:45:49.382Z INFO webdriver: RESULT null
[0-0] 2024-08-21T19:45:49.383Z INFO webdriver: COMMAND getWindowHandles()
[0-0] 2024-08-21T19:45:49.383Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/window/handles
[0-0] 2024-08-21T19:45:49.390Z INFO webdriver: RESULT [
[0-0]   '7C780A43FF9E5FEB981EFF9A6B2048E7',
[0-0]   '55598093E4652C509BEB5F7F0E406191'
[0-0] ]
[0-0] 2024-08-21T19:45:49.391Z INFO webdriver: COMMAND switchToWindow("<Screenshot[base64]>")
[0-0] 2024-08-21T19:45:49.391Z INFO webdriver: [POST] http://localhost:49642/session/6460d5231db5a019094c191847906560/window
[0-0] 2024-08-21T19:45:49.391Z INFO webdriver: DATA { handle: '7C780A43FF9E5FEB981EFF9A6B2048E7' }
[0-0] 2024-08-21T19:45:49.403Z INFO webdriver: RESULT null
[0-0] 2024-08-21T19:45:49.403Z INFO webdriver: COMMAND getUrl()
[0-0] 2024-08-21T19:45:49.404Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/url
[0-0] 2024-08-21T19:45:49.412Z INFO webdriver: RESULT https://www.google.com/
[0-0] 2024-08-21T19:45:49.413Z INFO webdriver: COMMAND getTitle()
[0-0] 2024-08-21T19:45:49.413Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/title
[0-0] 2024-08-21T19:45:49.423Z INFO webdriver: RESULT Google
[0-0] 2024-08-21T19:45:49.424Z INFO webdriver: BIDI COMMAND script.callFunction {"functionDeclaration":"<Function[17 bytes]>","awaitPromise":false,"arguments":[],"target":{"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}}
[0-0] 2024-08-21T19:45:49.431Z INFO webdriver: COMMAND switchToWindow("<Screenshot[base64]>")
[0-0] 2024-08-21T19:45:49.431Z INFO webdriver: [POST] http://localhost:49642/session/6460d5231db5a019094c191847906560/window
[0-0] 2024-08-21T19:45:49.431Z INFO webdriver: DATA { handle: '55598093E4652C509BEB5F7F0E406191' }
[0-0] 2024-08-21T19:45:49.441Z INFO webdriver: RESULT null
[0-0] 2024-08-21T19:45:49.441Z INFO webdriver: COMMAND getUrl()
[0-0] 2024-08-21T19:45:49.441Z INFO webdriver: [GET] http://localhost:49642/session/6460d5231db5a019094c191847906560/url
[0-0] 2024-08-21T19:45:49.450Z INFO webdriver: RESULT https://example.com/
[0-0] 2024-08-21T19:45:49.454Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:49.474Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:49.983Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:50.475Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:50.977Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:51.479Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:51.985Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:52.475Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:52.983Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:53.475Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:53.985Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:54.481Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:54.987Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:55.481Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:55.981Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:56.481Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:56.989Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:57.480Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:57.987Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:58.473Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] 2024-08-21T19:45:58.974Z INFO webdriver: BIDI COMMAND browsingContext.locateNodes {"locator":{"type":"css","value":"h1"},"context":"7C780A43FF9E5FEB981EFF9A6B2048E7"}
[0-0] Error in "My Login application.should switch to new window and get text"
Error: Can't call getText on element with selector "h1" because element wasn't found
    at implicitWait (C:\Users\user\Desktop\test\node_modules\webdriverio\build\index.js:49:13)
    at Element.elementErrorHandlerCallbackFn (C:\Users\user\Desktop\test\node_modules\webdriverio\build\index.js:6569:23)
    at Element.wrapCommandFn (C:\Users\user\Desktop\test\node_modules\@wdio\utils\build\index.js:1368:23)
    at async Context.<anonymous> (file:///C:/Users/user/Desktop/test/test/specs/test.e2e.js:19:22)
    at Context.executeAsync (C:\Users\user\Desktop\test\node_modules\@wdio\utils\build\index.js:1488:20)
    at Context.testFrameworkFnWrapper (C:\Users\user\Desktop\test\node_modules\@wdio\utils\build\index.js:1559:14)
[0-0] 2024-08-21T19:45:59.551Z INFO webdriver: COMMAND deleteSession()
[0-0] 2024-08-21T19:45:59.551Z INFO webdriver: [DELETE] http://localhost:49642/session/6460d5231db5a019094c191847906560
[0-0] 2024-08-21T19:45:59.608Z INFO webdriver: RESULT null
[0-0] 2024-08-21T19:45:59.608Z INFO webdriver: Kill driver process with PID 10716
[0-0] FAILED in chrome - file:///C:/Users/user/Desktop/test/test/specs/test.e2e.js
2024-08-21T19:45:59.732Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2024-08-21T19:45:59.733Z INFO @wdio/cli:launcher: Run onComplete hook

Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:17

2024-08-21T19:45:59.734Z INFO @wdio/local-runner: Shutting down spawned worker
2024-08-21T19:45:59.997Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2024-08-21T19:45:59.997Z INFO @wdio/local-runner: shutting down
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by Harsh-Personal** [comment]
For now after creating new window add `browser.browsingContextReload({context: newWindowHandle})` for e.g. it reloads context and bring context to your new window

`     await browser.newWindow(url, {
        windowName: newWindowName,
        windowFeatures: 'resizable,scrollbars=yes,status=1',
      })
      const newWindowHandle = await browser.getWindowHandle()
      await browser.browsingContextReload({context: newWindowHandle})
      reporter.addStep(`Created new window with handle ${newWindowHandle} and name ${newWindowName} with url ${url}`)
      return newWindowHandle`

**Comment by christian-bromann** [comment]
This got fixed in https://github.com/webdriverio/webdriverio/pull/13659 and released in `v9.1.1`. Please let me know if you still experience the issue.
